go-controller/pkg/factory/factory.go

```diff
@@ -239,7 +239,9 @@ func NewOVNKubeControllerWatchFactory(ovnClientset *util.OVNKubeControllerClient
 		stopChan:             make(chan struct{}),
 	}
 
-	if config.OVNKubernetesFeature.EnableMultiNetwork && !config.OVNKubernetesFeature.EnableInterconnect {
+	if config.OVNKubernetesFeature.EnableMultiNetwork &&
+		config.OVNKubernetesFeature.EnablePersistentIPs &&
```

Comment:
I can see you are turning off the initialization totally

I am not familiar with the feature so disregard me but I'd expect something like:
```
        if config.OVNKubernetesFeature.EnableMultiNetwork {
		cm.secondaryNetClusterManager, err = newSecondaryNetworkClusterManager(ovnClient, wf, recorder)
		if err != nil {
			return nil, err
		}
	}

	if config.OVNKubernetesFeature.EnableEgressIP {
		cm.eIPC = newEgressIPController(ovnClient, wf, recorder)
	}
```
this to be your entry point which should be shut off or is that not that case here?

Reply:
I think I'm missing something - what I'm trying to do it to prevent the listeners from being started if the CRD is not deployed. 

I unfortunately miss the point of your snippet @tssurya .

Reply:
right I'd expect the entire feature to switched off, but I only see the listeners prevented from being started, is that indirectly translating to "feature off" ?

Reply:
flags are to switch on/off entire feature; not to switch off if CRD is not available :) since the expectation is the CRD should already be installed in a cluster.

Reply:
The feature should be configurable now, and off by default. We do enable via the kind script always.

Still, I've checked it can tolerate the absence of the CRD by removing it (day1) and attempting to use the feature - an error is logged.

Reply:
Done.